src/cluster.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+
+#[derive(Debug, Clone)]
+pub struct NodeMeta {
```

Comment:
My idea was each server will get it config from cluster config. Thought of moving `ServerConfig` here. However thats a big change (can be part of another PR). For now i have added `NodeMeta` is just a place holder.